### PR TITLE
feat: allow configuring cors origins

### PR DESCRIPTION
**Pull request by jithin1987**
## Context

Allows configuring CORS origins 

## Objective

This will allow user to configure which all Origins can connect to SD API via Ajax.

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -198,3 +198,5 @@ ecosystem:
     badges: https://img.shields.io/badge/build-{{status}}-{{color}}.svg
     # Default registry to pull build containers from. Uses Docker Hub if nothing/empty string is provided
     dockerRegistry: ""
+    # Extra origins allowed to do CORS to API
+    allowCors: []
```
Comment (d2lam) [inline config/default.yaml]: I think we should set the default value to be same as ecosystem.ui

File: lib/server.js [hunk 2]
```diff
@@ -62,10 +62,15 @@ function prettyPrintErrors(request, reply) {
...
+
+    if (Array.isArray(config.ecosystem.allowCors)) {
+        corsOrigins = corsOrigins.concat(config.ecosystem.allowCors);
+    }
+
     const cors = {
-        origin: [
-            config.ecosystem.ui
-        ],
+        origin: corsOrigins,
```
Comment (d2lam) [inline lib/server.js]: and here just set it to config.ecosystem.allowCors

File: config/default.yaml (hunk 1 above)
Comment (jithin1987) [reply config/default.yaml]: ecosystem.ui is already in by default. This for extra origins. I didn't want users to repeat the ui value here again.

File: lib/server.js (hunk 2 above)
Comment (jithin1987) [reply lib/server.js]: This way we will be forcing users to repeat value of ecosystem.ui again here. That's just duplication and won't be backward compatible.

**Comment by jithin1987** [comment]
@minz1027 added unit test

**Pull request merged**